docs/source/cpp/streaming_execution.rst

```diff
@@ -175,9 +175,607 @@ their completion::
     // alive until this future is marked finished.
     Future<> complete = plan->finished();
 
+Constructing ``ExecNode`` using Options
+=======================================
+
+Using the execution plan we can construct varioud execution queries. 
+To construct such queries, we have provided a set of containers or 
+referred as :class:`ExecNode` s. These nodes provide the ability to 
+construct operations like filtering, projection, join, etc. 
+
+This is the list of :class:`ExecutionNode` s exposed;
+
+1. :class:`SourceNode`
+2. :class:`FilterNode`
+3. :class:`ProjectNode`
+4. :class:`ScalarAggregateNode`
+5. :class:`SinkNode`
+6. :class:`ConsumingSinkNode`
+7. :struct:`OrderBySinkNode`
+8. SelectK-SinkNode
```

Comment:
If some are undocumented, we should add the documentation so we can link them.

That said, I'm not sure how useful the API docs for nodes are, it might even be confusing.

Reply:
May be we can say about node operation without saying the node names. For instance join operation, filter operation, etc. I think it’s important to expose the available operations ? WDYT?

Reply:
I think we should just refer to them by their factory names, `"scan"`, `"filter"`, etc. As said below I don't think it really helps to link to the API docs for a node class.

Reply:
Yes, this would simplify and I can also avoid mentioning the nodes and rather talk about the operation and its options.

Reply:
Nice suggestion, table is neater and link to additional docs. I will update the docs.